### ISSUE TITLE
Build error on make project

### ISSUE THREAD

**Issue by eugeny-stoyka** [grpc/grpc#15576]
### **What version of gRPC and what language are you using?**
latest version

### **What operating system (Linux, Windows, …) and version?**
Archlinux last version

### **What runtime / compiler are you using (e.g. python version or version of gcc)**
gcc 8.1.0

### **What did you do?**
I run **make** command from root of the project directory

### **What did you see instead?**
```
[MAKE]    Generating cache.mk
[CXX]     Compiling src/core/lib/surface/channel.cc
src/core/lib/surface/channel.cc: In function ‘grpc_channel* grpc_channel_create_with_builder(grpc_channel_stack_builder*, grpc_channel_stack_type)’:
src/core/lib/surface/channel.cc:102:38: error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘grpc_channel’ {aka ‘struct grpc_channel’} with no trivial copy-assignment; use assignment or value-initialization instead [-Werror=class-memaccess]
   memset(channel, 0, sizeof(*channel));
                                      ^
src/core/lib/surface/channel.cc:60:8: note: ‘grpc_channel’ {aka ‘struct grpc_channel’} declared here
 struct grpc_channel {
        ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [Makefile:2853: /home/eugeny/Programming/C++/grpc/objs/opt/src/core/lib/surface/channel.o] Error 1
```

I think it's problem from version of gcc, isn't it?

Thanks!

**Comment by apolcyn** [comment]
I believe `-Werror=class-memaccess` was included in gcc-8 as a part of `-Wall`.... https://github.com/grpc/grpc/pull/15532 is a WIP PR that should fix this error

**Comment by eugeny-stoyka** [comment]
Do you know when it will be fixed? in a week or more?

**Comment by sab24** [comment]
It compiled with make CXXFLAGS="-Wno-error=class-memaccess" CFLAGS="-Wno-error=class-memaccess"

**Comment by mrngm** [comment]
Following steps from https://grpc.io/docs/quickstart/php.html#build-from-source-with-grpc-c-core-library gives the same error:
```make
$ git log --decorate=short | head -n1
commit d2c7d4dea492b9a86a53555aabdbfa90c2b01730 (HEAD, tag: v1.15.0)
$ make
[..]
[CXX]     Compiling src/core/lib/surface/channel.cc
src/core/lib/surface/channel.cc: In function ‘grpc_channel* grpc_channel_create_with_builder(grpc_channel_stack_builder*, grpc_channel_stack_type)’:
src/core/lib/surface/channel.cc:103:38: error: ‘void* memset(void*, int, size_t)’ clearing an object of type ‘grpc_channel’ {aka ‘struct grpc_channel’} with no trivial copy-assignment; use assignment or value-initialization instead [-Werror=class-memaccess]
   memset(channel, 0, sizeof(*channel));
                                      ^
src/core/lib/surface/channel.cc:61:8: note: ‘grpc_channel’ {aka ‘struct grpc_channel’} declared here
 struct grpc_channel {
        ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [Makefile:2913: /home/mrngm/local-scm/grpc/objs/opt/src/core/lib/surface/channel.o] Error 1
```

`grpc` is checked out on `v1.15.0` (see above), OS is Arch Linux, furthermore:
```
$ gcc --version
gcc (GCC) 8.2.1 20180831
$ g++ --version
g++ (GCC) 8.2.1 20180831
```

Neither `clang` nor `clang++` are installed.

**Comment by DexHunter** [comment]
Hi! I am installing grpc through `npm install` but it uses `node-pre-gyp` but I am not sure how to add the flags for gyp build? Any helps? Thanks in advance.

Following is an attachment of error:
![screenshot_20181109_113726](https://user-images.githubusercontent.com/6930518/48241966-faeabf00-e413-11e8-8496-63dadbcd55ea.png)

**Comment by Tsangares** [comment]
@sab24 Thank you!
Worked for me:
    export CC=/usr/bin/clang
    export CXX=/usr/bin/clang++
    npm install --build-from-source grpc

**Comment by Tsangares** [comment]
@DexHunter For me I had clang already installed. `sudo pacman -S clang`, then I just used 

    export CC=/usr/bin/clang
    export CXX=/usr/bin/clang++

before trying to install

**Comment by Bug-Reaper** [comment]
@Tsangares 

Thank you so much. Been dealing with GRPC issues foreverrr.

**Comment by nicolasnoble** [comment]
I've fixed gcc 8 issues linked with memset recently. This is available on the master branch and the 1.20.x branch. This one should (finally) be fixed.
